system/jlib/jtrace.cpp

```diff
@@ -427,31 +427,34 @@ class CSpan : public CInterfaceOf<ISpan>
         tracerName.set(nameOfTracer);
     }
 
-    void init()
+    void init(SpanFlags flags)
     {
         bool createLocalId = !isEmptyString(hpccGlobalId);
         if (createLocalId)
             hpccLocalId.set(ln_uid::createUniqueIdString().c_str());
 
-        auto provider = opentelemetry::trace::Provider::GetTracerProvider();
+        //we don't always want to create trace/span IDs
```

Comment:
Does this preserve trace-ids if this flags are false?

Reply:
> Does this preserve trace-ids if this flags are false?

Good point, opentel remote/local parentSpans won't have a span to associate with...

Reply:
@ghalliday see my latest commit. 
If a serverspan contains a remote parent, and we're not generating trace/span IDs we'll report the remote parent Trace and span ID:

```json
{"Type":"Server","Name":"propegatedServerSpan","GlobalID":"IncomingUGID","CallerID":"IncomingCID","LocalID":"JDaYP6r3d4MJEBvgUALsZu","ParentTraceID":"beca49ca8f3138a2842e5cf21402bfff","ParentSpanID":"4b960b3e4647da3f"}
```

If we're not generating trace/spans ids, then internal/client spans could contain a local parent, but that parent would likely not have a trace/span id either, but the current commit attempts to report the local parent trace and span id anyway. I'm not sure if this is what you envisioned.

Reply:
or, the child span can silently inherit trace id from the parent

Reply:
Latest amended commit simply inherits traceID from parent when no parent otel span available:

sample output for no traceID generated mode serverSpan(propagated parent)->clientSpanserverchild->internalSpan->internalSpan2

```json
15:07:40.802583 27843 Span end: {"Type":"Internal","Name":"internalSpan2","TraceID":"beca49ca8f3138a2842e5cf21402bfff","ParentSpanID": ""}
15:07:40.802618 27843 Span end: {"Type":"Internal","Name":"internalSpan","TraceID":"beca49ca8f3138a2842e5cf21402bfff","ParentSpanID": ""}
15:07:40.802628 27843 Span end: {"Type":"Client","Name":"clientSpanserverchild","TraceID":"beca49ca8f3138a2842e5cf21402bfff","ParentSpanID": ""}
15:07:40.802637 27843 Span end: {"Type":"Server","Name":"propegatedServerSpan","GlobalID":"IncomingUGID","CallerID":"IncomingCID","LocalID":"JDbF4xnv7LSWDV4Eug1SpJ","TraceID":"beca49ca8f3138a2842e5cf21402bfff","ParentSpanID":"4b960b3e4647da3f"}
```